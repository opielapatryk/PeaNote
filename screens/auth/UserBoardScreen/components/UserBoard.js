import React, { useEffect, useState } from 'react';
import { Text, TextInput, View, Pressable, Keyboard,  TouchableWithoutFeedback,Image,Dimensions } from 'react-native';
import {styles} from '../../../../assets/styles/styles'
import { useDispatch,useSelector } from 'react-redux';
import { setMyimage } from '../../../../store/settings/settingsSlice';
import auth, { firebase } from '@react-native-firebase/auth';
import { useFocusEffect } from '@react-navigation/native';
import firestore from '@react-native-firebase/firestore';
import * as FileSystem from 'expo-file-system';
import { sendFriendRequest } from '../functions/sendFriendRequest';

const UserBoard = ({ route, navigation }) => {
  const { friendEmail,name,oldnickname} = route.params;
  const dispatch = useDispatch()
  const { myimage } = useSelector((state) => state.settings);
  const { message } = useSelector((state) => state.login);
  const [description, newDescription] = useState('I love Peanotes!');
  const EMAIL = auth().currentUser.email

  const downloadImage = async (email) => {
    const imgDir = FileSystem.cacheDirectory + 'images/';
    const imgFileUri = imgDir + email;
    const imgUrl = await firebase.storage().ref(email).getDownloadURL() 

    // Checks if img directory exists. If not, creates it
    async function ensureDirExists() {
      const dirInfo = await FileSystem.getInfoAsync(imgDir);
      if (!dirInfo.exists) {
        await FileSystem.makeDirectoryAsync(imgDir, { intermediates: true });
      }
    }

    // Returns URI to our local img file
    // If our img doesn't exist locally, it downloads it
    async function getSingleImg() {
      await ensureDirExists();

      const fileUri = imgFileUri;
      const fileInfo = await FileSystem.getInfoAsync(fileUri);

      if (!fileInfo.exists) {
        await FileSystem.downloadAsync(imgUrl, fileUri);
      }

      dispatch(setMyimage(fileUri));
    }
    
    getSingleImg()
  }


  useFocusEffect(
    React.useCallback(() => {
      const getEmailByUsername = async () =>{
        const usernameSnapshot = await firestore()
          .collection('users')
          .where('username', '==', friendEmail)
          .get();

          newDescription(usernameSnapshot.docs[0].data().description)

        return usernameSnapshot.docs[0].data().email
      }

      getEmailByUsername().then(emailByUsername => {
        downloadImage(emailByUsername)
      });


      return ()=>{
        downloadImage(EMAIL)
      }
    }, [])
  );


  return (
    <TouchableWithoutFeedback 
    onPress={() => Keyboard.dismiss()}>
      <View style={{flex: 1,
    backgroundColor: '#FFF',
    justifyContent:"space-between",}}>
      <View style={{alignItems:'center',backgroundColor:'white'}}>
        <View style={styles.ProfilePicParent}>
        {myimage && <Image source={{uri: myimage}} style={styles.ProfilePic}/>}
       
        
        </View>

            <Text style={{textAlign:'center',fontStyle:'italic'}}>{description}</Text>

        <Pressable style={[styles.deleteAccountButton,{borderTopWidth:.17,borderColor:'lightgrey', marginTop:30,padding:10}]} onPress={()=>sendFriendRequest(dispatch,friendEmail,navigation)}><Text style={styles.removeFriendText}>{message?message:'ADD FRIEND'}</Text></Pressable>
        
        
      </View>

      <View>
        

      </View>
      
      </View>
    </TouchableWithoutFeedback>
  );
};

export default UserBoard;
