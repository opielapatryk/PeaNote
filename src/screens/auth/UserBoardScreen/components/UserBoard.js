import React, { useState } from 'react';
import { Text, View, Pressable, Keyboard,  TouchableWithoutFeedback,Image } from 'react-native';
import {styles} from '../../../../../assets/styles/styles'
import { useDispatch,useSelector } from 'react-redux';
import { sendFriendRequest } from '../functions/sendFriendRequest';
import { useDescription } from './useDescription';

const UserBoard = ({ route, navigation }) => {
  const { friendEmail,name,oldnickname} = route.params;
  const dispatch = useDispatch()
  const { friendimage } = useSelector((state) => state.settings);
  const { message } = useSelector((state) => state.login);
  const [invited, setInvited] = useState(false)

  return (
    <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
      <View style={{flex: 1,backgroundColor: '#FFF',justifyContent:"space-between"}}>
        <View style={{alignItems:'center'}}>
          <View style={styles.ProfilePicParent}>
            {friendimage && <Image source={{uri: friendimage}} style={styles.ProfilePic}/>}
          </View>

          <Text style={{textAlign:'center',fontStyle:'italic'}}>{useDescription(friendEmail)}</Text>

          <Pressable style={styles.addFriendButton} onPress={()=>sendFriendRequest(dispatch,friendEmail,navigation)}>
          <Text style={styles.removeFriendText}>{message?message:(invited?'REMOVE REQUEST':'ADD FRIEND')}</Text>
          </Pressable>  
       </View>
      </View>
    </TouchableWithoutFeedback>
  );
};

export default UserBoard;