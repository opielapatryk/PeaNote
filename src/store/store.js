import { configureStore } from '@reduxjs/toolkit'
import boardReducer from './notes/boardSlice'
import settingsReducer from './settings/settingsSlice'

export default configureStore({
    reducer:{
        board:boardReducer,
        settings:settingsReducer
    }
})