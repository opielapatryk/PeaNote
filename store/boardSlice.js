import { createSlice } from "@reduxjs/toolkit";

const initialState = {
    notes:[],
    isInput:true,   
    history:[]
}

export const boardSlice = createSlice({
    name:'board',
    initialState,
    reducers:{
        showInput: (state) =>{
            return {...state, isInput: true};
        },
        hideInput: (state) =>{
            return {...state, isInput: false};
        },
        addNote: (state, action) => {
            return {...state, notes:[...state.notes, action.payload],history: [...state.notes]}
        },
        removeNote: (state, action) => {
            return {
              ...state,
              notes: state.notes.filter(note => note.id !== action.payload),
              history: [...state.notes]
            };
          },
          changeInfo: (state, action) => {
            const prevText = state.history.find((note) => note.id === action.payload)?.text || "";
            return {
                ...state,
                notes: state.notes.map((note) => {
                    if (note.id === action.payload) {
                        return {
                            ...note,
                            isInfo: !note.isInfo,
                            text: note.isInfo ? prevText : 'click again to delete note',
                        };
                    }
                    return note;
                }),
                history: [...state.notes]
            };
        }     
    }
})

export const {showInput, addNote, removeNote, hideInput,changeInfo} = boardSlice.actions
export default boardSlice.reducer