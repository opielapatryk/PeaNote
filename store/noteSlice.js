import {createSlice } from '@reduxjs/toolkit'
const initialState = {
    id: 0,
    text:'',
    isNote:true,
    isInfo: false
}
export const noteSlice = createSlice({
    name:'note',
    initialState,
    reducers:{
        showInfo: (state)=>{
            return {...state,isInfo: true,text:'Click againt to delete note'}
        },
        setText: (state,action)=>{
            return {...state,text:action.payload}
        },
        hideInfo: (state)=>{
            return {...state,isInfo: false,text:initialState.text}
        },
        removeNote: (state)=>{
            return {...state,isNote: false}
        },
        isNoteTrue: (state)=>{
            return {...state,isNote: true}
        }
        
    }
})

export const {showInfo, hideInfo,removeNote,setText, isNoteTrue} = noteSlice.actions;

export default noteSlice.reducer