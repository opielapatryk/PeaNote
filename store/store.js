import { configureStore } from '@reduxjs/toolkit'
import boardReducer from './notes/boardSlice'

export default configureStore({
    reducer:{
        board:boardReducer
    }
})