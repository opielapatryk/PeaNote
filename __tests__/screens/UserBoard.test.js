import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';  
import store from '../../src/store/store'; 
import { SafeAreaProvider} from 'react-native-safe-area-context';
import UserBoard from '../../src/screens/auth/UserBoardScreen/components/UserBoard';

jest.mock('@react-native-firebase/auth', () => ({
    signInWithEmailAndPassword: jest.fn(),
    createUserWithEmailAndPassword: jest.fn(),
    signOut: jest.fn(),
}));

jest.mock('@react-native-google-signin/google-signin', () => ({
    signIn: jest.fn(),
}));


jest.mock('@invertase/react-native-apple-authentication', () => ({
  AppleButton: jest.fn(),
}));

jest.mock('@react-native-firebase/firestore', () => ({
    collection: jest.fn(),
    doc: jest.fn(),
    get: jest.fn(),
}));


it('Login screen renders correctly.', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <SafeAreaProvider>
        <UserBoard />
      </SafeAreaProvider>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});