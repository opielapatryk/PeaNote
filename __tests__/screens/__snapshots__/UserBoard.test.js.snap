// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBoard screen renders correctly. 1`] = `null`;
